experimental/iree/run_benchmarks.py

```diff
@@ -32,48 +35,155 @@
+      f"--task_topology_cpu_ids={task_topology_cpu_ids}",
       "--device=local-task",
       "--function=main",
-      f"--expected_output=@{output_npy}",
-      f"--expected_f32_threshold={atol}",
-      f"--expected_f16_threshold={atol}",
-      f"--expected_f64_threshold={atol}",
   ]
 
-  inputs_dir = artifact_dir / "inputs_npy"
-  num_inputs = len(list(inputs_dir.glob("*.npy")))
-  for i in range(num_inputs):
-    command.append(f"--input=@{inputs_dir}/input_{i}.npy")
+  inputs_dir = artifacts_dir / "inputs_npy"
+  inputs = get_directory_names(target_device, inputs_dir)
+  for input in inputs:
+    parameters.append(f"--input=@{inputs_dir}/{input}")
+
+  return parameters
+
+
+def generate_accuracy_check_command(target_device: def_types.DeviceSpec,
+                                    artifacts_dir: pathlib.Path,
+                                    iree_run_module_path: pathlib.Path,
+                                    atol: float,
+                                    task_topology_cpu_ids: str) -> str:
+  output_npy = artifacts_dir / "outputs_npy" / "output_0.npy"
+  command = [str(iree_run_module_path)] + get_common_command_parameters(
+      target_device, artifacts_dir, task_topology_cpu_ids) + [
+          f"--expected_output=@{output_npy}",
+          f"--expected_f32_threshold={atol}",
+          f"--expected_f16_threshold={atol}",
+          f"--expected_f64_threshold={atol}",
+      ]
+  return " ".join(command)
+
+
+def benchmark_on_x86(target_device: def_types.DeviceSpec,
+                     benchmark: def_types.BenchmarkCase,
+                     artifacts_dir: pathlib.Path,
+                     iree_run_module_path: pathlib.Path,
+                     iree_benchmark_module_path: pathlib.Path,
+                     task_topology_cpu_ids: str,
+                     verbose: bool) -> Dict[str, Any]:
+  # Check accuracy.
+  atol = benchmark.verify_parameters["absolute_tolerance"]
+  command = generate_accuracy_check_command(target_device, artifacts_dir,
+                                            iree_run_module_path, atol,
+                                            task_topology_cpu_ids)
+
+  try:
+    output = subprocess.run(command,
+                            shell=True,
+                            check=True,
+                            capture_output=True)
+    if verbose:
+      print(output.stdout.decode())
+    is_accurate = True
+  except subprocess.CalledProcessError as e:
+    print(f"Error running command: {e}")
+    is_accurate = False
+
+  # Run benchmark.
+  command = [str(iree_benchmark_module_path)] + get_common_command_parameters(
+      target_device, artifacts_dir,
+      task_topology_cpu_ids) + ["--print_statistics"]
+  metrics = benchmark_lib.run_benchmark_command(" ".join(command), verbose)
+  metrics["accuracy"] = is_accurate
+  return metrics
 
-  command_str = " ".join(command)
-  print(f"Running command: {command_str}")
+
+def benchmark_on_arm(target_device: def_types.DeviceSpec,
```

Comment:
Maybe `benchmark_on_android`?

```suggestion
def benchmark_on_android(target_device: def_types.DeviceSpec,
```

Reply:
Replaced